### PR TITLE
Create EOL comms UI overview and template page

### PR DESCRIPTION
**Pull request by tabathadelane**
### ✨ Summary

- Add EOL template for individual announcements
- Add EOL page for overview of all announcements `/eol`
- Exclude from verify-mdx, and translations
- Add sample pages
- Add publish date to frontmatter
- Create tabe with sortable column headers
- Sort by publish date OR eol date and display both on feed page
- Note: I focused on functionality only. Design polish is out-of-scope

### 🌠 Screen capture

https://github.com/user-attachments/assets/cb3363a3-de22-444f-85f8-4cdc67ba8ed4

### 🔮 Engineering checklist

- [ ] Relevant documentation including README's or Confluence are updated to reflect these changes
- [ ] Code changes are self-documenting or clearly commented/documented
- [ ] Changes that effect tech writers have been communicated to tech docs team
- [ ] New components created for use in MDX files have been serialized/deserialized
- [ ] New tests are added where applicable and total coverage remains >70%
- [ ] No new warnings or errors where added to the browser or build console
- All downstream dependencies for DaaS have been considered for negative impacts
  - [ ] What's new JSON endpoints
  - [ ] Release notes JSON & EOL pages
- [ ] The build preview has been checked for expected functionality on desktop browsers
- [ ] The build preview has been checked for expected functionality on mobile browsers
- [ ] Synthetics tests have been considered for any changes that might trigger false positives

### CODE REVIEW

File: src/pages/eol.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  const now = useMemo(() => new Date(), []);
+  const { queryByEOLDate, queryByPublishDate } = data;
+
+  const tableHeaders = [
+    { label: 'Published', contentId: 'publishDate', sort: true },
+    { label: 'EOL Effective', contentId: 'eolDate', sort: true },
+    { label: '', contentId: 'details', sort: false },
+  ];
+
+  const shapePostDate = (posts) => {
```
Comment (tabathadelane) [inline src/pages/eol.js:32]: this is typo

File: src/content/eol/2024/07/images/.gitkeep
Comment (clarkmcadoo) [inline src/content/eol/2024/07/images/.gitkeep:1]: what's this?

Comment (tabathadelane) [reply src/content/eol/2024/07/images/.gitkeep:1]: the images folder?

Comment (tabathadelane) [reply src/content/eol/2024/07/images/.gitkeep:1]: I assume we want this to follow the same pattern as whats-new. It was requested we had some samples in place for when it comes time to add content. I did not want to actually duplicate or add any unnecessary images to these files, but the whats-new convention is to create new directories as needed with a `.gitkeep` while they are empty.

File: src/pages/eol.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/react';
+import PageTitle from '../components/PageTitle';
```
Comment (clarkmcadoo) [inline src/pages/eol.js:4]: let's keep all the 3rd party imports at the top then follow those by local imports

Comment (clarkmcadoo) [reply src/pages/eol.js:4]: aka - move this to line 12

File: src/templates/eolAnnouncement.js [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+  );
+};
+
+EolTemplate.propTypes = {
+  data: PropTypes.object.isRequired,
+  location: PropTypes.object.isRequired,
+  pageContext: PropTypes.object.isRequired,
+};
+
+export const pageQuery = graphql`
```
Comment (clarkmcadoo) [inline src/templates/eolAnnouncement.js:140]: `nit` does this need to be exported?

File: src/templates/eolAnnouncement.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        learnMoreLink
+      }
+      fields {
+        fileRelativePath
+      }
+    }
+  }
+`;
+
+const DateIcon = ({ children, icon, dateString }) => {
```
Comment (clarkmcadoo) [inline src/templates/eolAnnouncement.js:163]: Can we make the `DateIcon` and `MetaLink` styled components vs new components? That would get rid of the need for propTypes for them and simplify the file

File: src/templates/eolAnnouncement.js (hunk 3 above)
Comment (tabathadelane) [reply src/templates/eolAnnouncement.js:140]: Yes. That is how it is used in the gatsby docs and I get an `undefined` error from the data in the props if i remove it

**Review by clarkmcadoo — APPROVED** [review]

**Pull request merged**